stdlib/camlinternalMod.ml

```diff
@@ -54,9 +54,10 @@ let rec update_mod shape o n =
       (* The optimisation below is invalid on bytecode since
          the RESTART instruction checks the length of closures.
          See PR#4008 *)
```

Comment:
The comment needs updating:
In bytecode, the RESTART instruction checks the size of closures.
Hence, the optimized case overwrite o n is valid only if o and n have the same size.
(See PR#4008.)
In native-code, the size of closures does not matter, so overwriting is possible
as soon as the size of n is less than that of o.
```

Reply:
Thanks for the suggestion, I've committed a slightly modified version of this.